### PR TITLE
docker-machine update to 0.5.0

### PR DESCRIPTION
**Pull request by vlesierse**
See issue #45352 

Docker Machine 0.5.0 has been released and this PR can be merged

### CODE REVIEW

File: Library/Formula/docker-machine.rb [hunk 1]
```diff
@@ -3,56 +3,20 @@
 class DockerMachine < Formula
   desc "Create Docker hosts locally and on cloud providers"
   homepage "https://docs.docker.com/machine"
-  url "https://github.com/docker/machine/archive/v0.4.1.tar.gz"
-  sha256 "f089657b2de7a3ce15374e69be3f654b0866f75eb077ca363f8a5933ccf51cda"
+  url "https://github.com/docker/machine/archive/v0.5.0-rc3.tar.gz"
+  sha256 "117cb0bfd0400e20831932c412796b1f2e28db239b472ee71e109d6d1595399d"
   head "https://github.com/docker/machine.git"
-
-  bottle do
```
Comment (DomT4) [inline Library/Formula/docker-machine.rb]: You don't need to remove this block on update FYI.

**Comment by karlhorky** [comment]
How would I extract this to make it into a tap for myself?

Do I need to also recursively add all dependencies to my repo as well? Like `requires "language/go"` and all of Go's dependencies, `depends_on "automake"` etc...

**Comment by bfontaine** [comment]
@karlhorky just put the file in your tap and you should be fine if you build from source.

**Comment by karlhorky** [comment]
@bfontaine Thanks! I think I have [the tap](https://github.com/karlhorky/homebrew-docker-machine) working.

What I did:

```bash
brew uninstall docker-machine
brew tap karlhorky/docker-machine
brew install karlhorky/docker-machine/docker-machine
brew link karlhorky/docker-machine/docker-machine
```

It installed Go and built from source.

**Comment by bfontaine** [comment]
@karlhorky You can do this in two steps only:
```
brew uninstall docker-machine
brew install karlhorky/docker-machine/docker-machine
```

**Comment by karlhorky** [comment]
Awesome, cheers. I'll remember that for next time.

File: Library/Formula/docker-machine.rb (hunk 1 above)
Comment (vlesierse) [inline Library/Formula/docker-machine.rb]: Ok, reverted this deletion

**Comment by vlesierse** [comment]
@DomT4 Docker Machine 0.5.0 has been release and if OK this PR can be accepted.

**Comment by mikemcquaid** [comment]
Why does it not use any Go resources any more?

**Comment by vlesierse** [comment]
@mikemcquaid the Makefile already takes care of that. The way of building docker machine has change with this release.

**Comment by jfchevrette** [comment]
:+1:

**Comment by bfontaine** [comment]
@jfchevrette Please don’t write “:+1:” comments unless you have something to add to the discussion; thanks!

**Comment by dunn** [comment]
How are the dependencies downloaded during the `make` step? Are they pinned to a specific release, or does it just download the `HEAD`?

**Comment by vlesierse** [comment]
Looks like it is using the `HEAD`. @nathanleclaire can you elaborate on this?

**Comment by databus23** [comment]
Looks like machine uses the experimental vendoring support from go 1.5:
https://github.com/docker/machine/blob/master/mk/main.mk#L9

That means the go build considers the `vendor/` directory when resolving dependencies. Thats where all the dependencies are placed in the machine repo.

**Comment by databus23** [comment]
Its definitely not downloading anything during build. All dependencies are vendored in the tarball. No problems regarding a reproducible build.

**Comment by vlesierse** [comment]
@databus23 thnx for the clarification...

**Comment by nathanleclaire** [comment]
Yep @databus23 has it correct.  Building using the vendored deps is how it's meant to build and if it doesn't, it's a bug.

**Comment by DomT4** [comment]
Is this good to merge? (Ideally a question answered by someone from upstream). Would be a relief to stop people filing new PRs for this every 20 minutes :stuck_out_tongue_winking_eye:.

**Comment by vlesierse** [comment]
@nathanleclaire? I know it work very well from my own branch :+1:

**Comment by jeroenj** [comment]
I've been using this change during the day today without any issues.

**Comment by nathanleclaire** [comment]
It looks and sounds like everything is good to go here but I am not super familiar with Homebrew and how the process is expected to work.

**Comment by DomT4** [comment]
Good enough for me.

Merged in https://github.com/Homebrew/homebrew/commit/8f615708184884e501bf5c16482c95eff6aea637. Thank you for your contribution to Homebrew @vlesierse; we appreciate it! :smiley_cat:

**Comment by nathanleclaire** [comment]
:+1:

**Pull request closed without merging**